llvm/lib/Target/DirectX/DXContainerGlobals.cpp

```diff
@@ -143,23 +145,35 @@ void DXContainerGlobals::addPipelineStateValidationInfo(
   SmallString<256> Data;
   raw_svector_ostream OS(Data);
   PSVRuntimeInfo PSV;
-  Triple TT(M.getTargetTriple());
   PSV.BaseData.MinimumWaveLaneCount = 0;
   PSV.BaseData.MaximumWaveLaneCount = std::numeric_limits<uint32_t>::max();
+
+  dxil::ModuleMetadataInfo &MMI =
+      getAnalysis<DXILMetadataAnalysisWrapperPass>().getModuleMetadata();
+  assert(MMI.EntryPropertyVec.size() != 0 ||
+         MMI.ShaderStage == Triple::Library);
```

Comment:
What stops us from getting here with zero entry points? I guess we'll have diagnosed that and errored out earlier? Similarly, what if there are multiple entry points? Is that only possible for library shaders?

Reply:
Good catch. Should be size == 1.
Multiple entry points or no entry point should be only possible for library shaders.